### PR TITLE
feat(gui): highlight active band button in spectrum overlay band panel

### PR DESCRIPTION
**Pull request by nonoo**
## Summary

Highlights the currently active band button in the panadapter spectrum overlay's floating band selection menu based on the active slice's frequency.

- When the active slice frequency changes or when a slice is attached/detached, the corresponding band button updates its checked state.
- Matches across standard HF bands, configured XVTR bands, and radio-declared band ranges using `BandSettings::bandForFrequency()`.
- Uses `QSignalBlocker` to update button checked state cleanly without triggering spurious click emissions.
- Inlines `wfSchemeName()` in `SpectrumWidget.h` so standalone overlay test targets link without pulling heavy rendering dependencies.

## Constitution principle honored

Principle II — The Radio Is Authoritative On Live State (overlay band button highlight reflects the active slice frequency and live band state from the radio model).
Principle IV — Every Contribution Is Clean-Room (code is clean-room, not derived from decompiled sources).

## Test plan

- [x] Local build passes (`cmake --build build`)
- [X] Behavior verified on a real radio if applicable
- [x] Existing tests pass (CI)
- [ ] Reproduction steps documented if user-reported bug

Verification details:
- Unit test suite: `spectrum_overlay_band_highlight_test` built cleanly and passed all 10 test assertions (standard HF bands, general coverage, transverter bands, and slice attachment/detachment).
- Full suite of repository static checks passed with `--strict` (engine boundary, test registration, network timeouts, theme seed, GLSL dialects, accessibility).

## Checklist

- [x] Commits are signed (`docs/COMMIT-SIGNING.md`)
- [x] No new flat-key `AppSettings` calls — use nested-JSON-under-one-key
      (Principle V)
- [x] Code is clean-room — not decompiled, disassembled, or
      reverse-engineered from a proprietary binary (Principle IV)
- [ ] All meter UI uses `MeterSmoother` (AGENTS.md convention)
- [ ] Documentation updated if user-visible behavior changed — `docs/` and the
      affected READMEs. **Not `CHANGELOG.md`**, which is a release-prep file a
      PR must not add to (AGENTS.md); describe it in the Summary above instead
- [ ] Security-sensitive changes reference a GHSA if applicable

### CODE REVIEW

File: src/gui/SpectrumOverlayMenu.cpp [hunk 1]
```diff
@@ -467,12 +467,18 @@ void SpectrumOverlayMenu::buildBandPanel()
...
+    m_bandBtnFreqs.clear();
+
+    const QString bandBtnStyle = ThemeManager::instance().resolve(
+        QStringLiteral(
+            "QPushButton { background: rgba(30, 40, 55, 220); "
+            "border: 1px solid {{color.background.2}}; border-radius: 3px; "
+            "color: {{color.text.primary}}; font-size: 11px; font-weight: bold; }"
+            "QPushButton:hover { background: rgba(0, 112, 192, 180); "
+            "border: 1px solid {{color.accent}}; }"
+            "QPushButton:checked { background: {{color.accent.dim}}; color: {{color.text.primary}}; "
```
Comment (jensenpat) [inline src/gui/SpectrumOverlayMenu.cpp]: **Blocker 1 — approval boundary:** this `:checked` state is a new visual/default UX behavior on an existing control. The PR has no linked issue, RFC, or maintainer ruling, while `GOVERNANCE.md` explicitly requires an RFC for visual-design changes and default UX behavior. Please link the approved decision, or link an accepted bug report that defines this highlight as the intended repair.

File: src/gui/SpectrumOverlayMenu.cpp [hunk 2]
```diff
@@ -467,12 +467,18 @@ void SpectrumOverlayMenu::buildBandPanel()
...
-    const QString bandBtnStyle =
-        "QPushButton { background: rgba(30, 40, 55, 220); "
-        "border: 1px solid #304050; border-radius: 3px; "
-        "color: #c8d8e8; font-size: 11px; font-weight: bold; }"
-        "QPushButton:hover { background: rgba(0, 112, 192, 180); "
-        "border: 1px solid #0090e0; }";
+    m_bandButtons.clear();
+    m_bandBtnFreqs.clear();
+
+    const QString bandBtnStyle = ThemeManager::instance().resolve(
```
Comment (jensenpat) [inline src/gui/SpectrumOverlayMenu.cpp]: **Blocker 2 — live theming:** `resolve()` turns these tokens into one-time literal QSS, and each button later receives that string through `setStyleSheet()`. It is therefore absent from ThemeManager's tracked-widget map and will retain the old theme after `themeChanged`. Keep this as an unresolved template and apply it to each button with `ThemeManager::applyStyleSheet()` (in both initial construction and the XVTR rebuild path); that API resolves per widget and reapplies on theme switches.

File: src/gui/SpectrumOverlayMenu.cpp [hunk 3]
```diff
@@ -3317,6 +3349,49 @@ void SpectrumOverlayMenu::applyTuningRangeToBandButtons()
...
+            if (entry.button) {
+                QSignalBlocker b(entry.button);
+                entry.button->setChecked(false);
+            }
+        }
+        return;
+    }
+
+    const double freq = m_slice->frequency();
+    QString activeBand = BandSettings::bandForFrequency(freq);
```
Comment (jensenpat) [inline src/gui/SpectrumOverlayMenu.cpp]: **Blocker 3 — region correctness:** `BandSettings::bandForFrequency()` delegates to the ARRL/US-only `BandDefs.h` table. `AGENTS.md` explicitly says new consumers of band edges must use the active `BandPlanManager`; otherwise the highlight disagrees with the selected regional plan (for example, 7.250 MHz is outside IARU Region 1's 40 m allocation but maps to `40m` here). Please route a plan-aware band lookup into the overlay and pin at least one non-US edge case.

**Review by jensenpat — CHANGES_REQUESTED** [review]
The standard-HF and slice attach/detach paths match the PR's stated intent, but this head is not merge-ready. There is no linked issue, RFC, or maintainer ruling for the new checked-state visual/UX behavior, and the implementation has two user-visible correctness gaps: it freezes token values at panel construction instead of following live theme changes, and it derives band edges from the US-only `BandDefs` table instead of the active regional band plan.

### Scope

| File/group | Claimed by title/body? | Verdict |
|---|---|---|
| `SpectrumOverlayMenu.{h,cpp}` | Yes | In scope, but blocked by the three findings below |
| `SpectrumWidget.{h,cpp}` | Yes — described as the standalone-test link seam | In scope; implementation relocation is coherent |
| `tests/spectrum_overlay_band_highlight_test.cpp`, `tests/tests.cmake` | Yes | In scope; focused coverage exists, but it does not cover regional-plan or declared-range behavior |

The PR's limited-scope claim holds up; I found no unrelated files or commits.

### Blockers

1. The new checked-state visual/default UX behavior has no approved RFC or maintainer ruling. `GOVERNANCE.md` says any visual-design change and any default-UX-behavior change require an RFC, while the PR has no linked issue at all. Link the approved decision (or an issue establishing this as a bug fix with intended behavior) before this can merge. See inline comment 1.
2. The new tokenized button styles are converted to literal QSS once with `resolve()` and then applied with `setStyleSheet()`. `ThemeManager::applyStyleSheet()` is the API that records the unresolved template and reapplies it on `themeChanged`; as written, switching themes leaves the band buttons, including the new active fill, painted with the previous theme. See inline comment 2.
3. `BandSettings::bandForFrequency()` reads `BandDefs.h`, whose table is explicitly ARRL/US-only. This is a new band-edge consumer, and `AGENTS.md` requires active-plan data through `BandPlanManager`; for example, 7.250 MHz is outside IARU Region 1's 40 m allocation but this code highlights the 40 m button. Use a plan-aware lookup and add a non-US regression test. See inline comment 3.

### Nits (non-blocking)

- The new test does not exercise the PR body's radio-declared-band-range claim. Add a declared `2m`/`440`/`23cm` range case while revising the focused test.
- The configured-XVTR test proves only the exact configured center; the implementation's undocumented +/-0.5 MHz heuristic is not evidence for the broader 'configured XVTR band' claim. Either document/justify that boundary or cover representative off-center frequencies.

### Verification

Reviewed exact head `593518a2bd3abd85e7ab34eb397159c2fa453cb0`, whose signed commit is a direct child of current `origin/main` (`fb3a7e3d424a91ebe2484e529d9db8d0d141910f`). `git merge-tree --write-tree origin/main HEAD` produced clean tree `f48b16c412e3a9f8c2e7fd857991b332af1b49e9`. All four reported GitHub checks are green and there were no pre-existing review threads. Locally, the exact CI hardcoded-colour ratchet passed with `unique_colours -1`, `total_references -9`, and `setstylesheet +0`; `check_theme_seed.py --strict`, engine-boundary, test-registration, network-timeout, shader-dialect, and focused accessibility checks also passed. I stopped before an expensive rebuild because the governance and behavioral blockers above already prevent merge.

File: src/gui/SpectrumOverlayMenu.cpp [hunk 4]
```diff
@@ -467,12 +467,18 @@ void SpectrumOverlayMenu::buildBandPanel()
...
+
+    const QString bandBtnStyle = ThemeManager::instance().resolve(
+        QStringLiteral(
+            "QPushButton { background: rgba(30, 40, 55, 220); "
+            "border: 1px solid {{color.background.2}}; border-radius: 3px; "
+            "color: {{color.text.primary}}; font-size: 11px; font-weight: bold; }"
+            "QPushButton:hover { background: rgba(0, 112, 192, 180); "
+            "border: 1px solid {{color.accent}}; }"
+            "QPushButton:checked { background: {{color.accent.dim}}; color: {{color.text.primary}}; "
+            "border: 1px solid {{color.accent}}; }"));
```
Comment (ten9876) [inline src/gui/SpectrumOverlayMenu.cpp]: **Blocker 1 — `{{color.text.primary}}` over a hardcoded dark background is unreadable in the light theme.**

The background on the line above stays `rgba(30, 40, 55, 220)`, but the text now resolves through the theme. In `resources/themes/default-light.json`:

```
color.text.primary = #1a2a3a
```

That is dark navy on dark navy. Computed contrast: **≈1.0:1** against the raw `rgb(30,40,55)`, and **≈1.5:1** composited at α=0.86 over a light panel. WCAG AA wants 4.5:1, and 11 px bold is below the 18.66 px threshold that would let it use the 3:1 large-text allowance. Both theme files define only a `root` scope, so nothing overrides `text.primary` for this widget.

**This pattern is not yours** — `SpectrumOverlayMenu.cpp:326` already pairs `{{color.text.primary}}` with a hardcoded `rgba(20, 30, 45, 240)` on the toggle arrow, and is equally broken in light mode today. The difference is scale: that is one arrow, this is the whole band grid plus both XVTR styles.

Two ways out, and the second is the smaller diff:

**(a) Move the background onto the theme too**, so the pair stays consistent:

```
"QPushButton { background: {{color.background.1}}; "
```

**(b) Keep the text hardcoded exactly as it was and add only what the feature needs** — which is the `:checked` rule, nothing else:

```suggestion
    const QString bandBtnStyle = ThemeManager::instance().resolve(
        QStringLiteral(
            "QPushButton { background: rgba(30, 40, 55, 220); "
            "border: 1px solid #304050; border-radius: 3px; "
            "color: #c8d8e8; font-size: 11px; font-weight: bold; }"
            "QPushButton:hover { background: rgba(0, 112, 192, 180); "
            "border: 1px solid #0090e0; }"
            "QPushButton:checked { background: {{color.accent.dim}}; color: #c8d8e8; "
            "border: 1px solid {{color.accent}}; }"));
```

(b) keeps this PR to its stated purpose and leaves the file no worse in light mode than it is now. The full migration is worth doing — see my note on the XVTR styles — just not folded in here.

File: src/gui/SpectrumOverlayMenu.cpp [hunk 5]
```diff
@@ -3022,19 +3038,25 @@ void SpectrumOverlayMenu::setXvtrBands(const QVector<XvtrBand>& bands)
...
+
+    const QString xvtrBtnStyle = ThemeManager::instance().resolve(
+        QStringLiteral(
+            "QPushButton { background: rgba(30, 40, 55, 220); "
+            "border: 1px solid {{color.background.2}}; border-radius: 3px; "
+            "color: {{color.accent.bright}}; font-size: 11px; font-weight: bold; }"
+            "QPushButton:hover { background: rgba(0, 112, 192, 180); "
+            "border: 1px solid {{color.accent}}; }"
+            "QPushButton:checked { background: {{color.accent.dim}}; color: {{color.text.primary}}; "
+            "border: 1px solid {{color.accent}}; }"));
```
Comment (ten9876) [inline src/gui/SpectrumOverlayMenu.cpp]: **Blocker 2 — the migration is undisclosed, changes two colours, and stops three-quarters of the way through.**

Resolving each substitution against `default-dark.json` (via its `primitives` map):

| Was | Now | Resolves to | Identity? |
|---|---|---|---|
| `#304050` | `{{color.background.2}}` | `#304050` | ✅ |
| `#c8d8e8` | `{{color.text.primary}}` | `#c8d8e8` | ✅ |
| `#0090e0` | `{{color.accent}}` | **`#00b4d8`** | ❌ |
| `#00d0ff` | `{{color.accent.bright}}` | **`#00c8f0`** | ❌ |

So the hover border and the XVTR label colour both shift, in the default theme, in a PR whose body says it highlights the active band button. Small shifts, and arguably toward the correct semantic colour — but a reader of the body has no way to know they happened.

The incompleteness is the other half. There is a fourth stylesheet with the identical six lines at `SpectrumOverlayMenu.cpp:3242` (the separate XVTR panel), and it is untouched. After this, the XVTR panel and the band panel disagree with each other in both themes.

I want to be clear this is not a "don't do it" — moving this file onto theme tokens is the direction the colour ratchet and `tools/check_theme_seed.py` exist to encourage, and someone should. The ask is to do it as its own change: all four stylesheets, the background moved along with the text so light mode works (blocker 1), and a body that says the colours shift. Then this PR is just the `:checked` rule and the highlight logic, which is a genuinely tidy diff.

File: src/gui/SpectrumOverlayMenu.cpp [hunk 6]
```diff
@@ -3317,6 +3349,49 @@ void SpectrumOverlayMenu::applyTuningRangeToBandButtons()
...
+    for (const auto& range : m_declaredBandRanges) {
+        if (range.lowHz > 0.0 && range.highHz > 0.0) {
+            const double lowMhz = range.lowHz / 1.0e6;
+            const double highMhz = range.highHz / 1.0e6;
+            if (freq >= lowMhz && freq <= highMhz) {
+                activeBand = range.name;
+                break;
+            }
+        }
+    }
```
Comment (ten9876) [inline src/gui/SpectrumOverlayMenu.cpp:3429]: **Finding 3 — this precedence is backwards relative to what clicking the button does.**

Here the declared-range loop runs *after* the XVTR loop and overwrites `activeBand`, so a declared range wins.

`MainWindow::selectBand()` — the handler on the other end of `bandSelected` — resolves the opposite way. `XvtrPolicy::resolveBandStackKey()` runs first, and the declared-band branch is guarded by `if (stackKey.isEmpty())`, so it only runs when no XVTR matched. The comment there states it as a rule rather than an accident:

> The same FIRST-wins rule means a user XVTR labelled e.g. "2m" out-ranks a gateway's declared "2m" … That is intentional and consistent with the XVTR-priority precedent already codified for native bands (#2342) — declared tokens do not override a same-named user XVTR.

On a radio that declares bands (an IC-9700 gateway, say) where the operator also has an overlapping XVTR configured, the button this highlights and the band a click actually selects are different. That is the one outcome a highlight feature must not produce.

Swapping the two loops so the XVTR check runs last — or giving the XVTR loop an early `return` on match — lines it up with `selectBand()`.

**Finding 4, same function** — `std::abs(freq - xvtr.rfFreqMhz) < 0.5` is an uncommented ±500 kHz window around a single point frequency, and the loop immediately below shows the shape that was available: `DeclaredBandRange` carries real `lowHz`/`highHz` bounds and gets a proper containment test. A transverter covering 144–148 MHz will highlight only within 500 kHz of its configured RF frequency, and two XVTRs under 1 MHz apart both match with first-wins arbitration. If `XvtrBand` has no range to test against, the constant at least wants a comment explaining what 0.5 is doing.

**Review by ten9876 — DISMISSED** [review]
## Issue fit

No linked issue, so I reviewed this against its own stated intent: highlight the band button matching the active slice's frequency. It does that, and several details are right — `updateActiveBandHighlight()` is called from all four places that can invalidate it (`buildBandPanel`, `setSlice`, `showBandPanelAt`, and the `frequencyChanged` connection), `m_bandButtons` is cleared alongside `m_bandBtnFreqs` before the panel rebuild so it cannot dangle the way that existing comment warns about, and `QSignalBlocker` keeps `setChecked()` from re-emitting `clicked`.

GOVERNANCE.md §"What requires an RFC" lists *"New feature areas substantially beyond the current scope"* — a checked-state highlight on existing buttons is not that, so no RFC is needed here. An issue would still have helped, because two of the four things in this diff are not in the body.

**Verified empirically** on macOS/arm64: clean build of the full app target, and `spectrum_overlay_band_highlight_test` passes (1.29 s).

## Scope

| Change | In the body? | Verdict |
|---|---|---|
| `m_bandButtons`, `updateActiveBandHighlight()`, `setCheckable(true)`, the `frequencyChanged` hookup | Yes | In scope |
| `QPushButton:checked` rules added to three stylesheets | Yes, implicitly | In scope — this is what the feature needs |
| **Hardcoded hex → `ThemeManager` tokens in three stylesheets** | **No** | **Undisclosed. Blockers 1 and 2** |
| `wfSchemeName()` moved to an `inline` header function | Yes | In scope, and a reasonable way to keep the test target light |
| `tests/spectrum_overlay_band_highlight_test.cpp` (+116), `tests.cmake` (+34) | Yes | In scope |

No `CHANGELOG.md` entry — correct.

## Blockers

**1. In the light theme these buttons become unreadable.** *(inline: `SpectrumOverlayMenu.cpp:477`)* `color:` moves from the hardcoded `#c8d8e8` to `{{color.text.primary}}`, but the background stays hardcoded `rgba(30, 40, 55, 220)`. In `resources/themes/default-light.json`, `color.text.primary` resolves to **`#1a2a3a`** — dark navy text on a dark navy panel. Measured contrast is **≈1.0:1** against the raw background and **≈1.5:1** composited over a light panel; WCAG AA wants 4.5:1, and 11 px bold does not qualify for the 3:1 large-text allowance. Both theme files define only a `root` scope, so there is no per-widget override that rescues it.

To be fair about provenance: **this PR did not invent the pattern.** `SpectrumOverlayMenu.cpp:326` already pairs `{{color.text.primary}}` with a hardcoded `rgba(20, 30, 45, 240)` on the toggle arrow, and has the same problem today. What changes here is the surface — one arrow button becomes the entire band grid plus both XVTR button styles, roughly twenty buttons, which is the difference between a latent wart and the band panel being unusable in light mode.

**2. The token migration is undisclosed, partial, and changes two colours in the dark theme too.** *(inline: `SpectrumOverlayMenu.cpp:3043`)* The body's four bullets do not mention it at all. Resolved against `default-dark.json`, two of the four substitutions are not identity:

| Was | Now | Dark value | Same? |
|---|---|---|---|
| `#304050` border | `{{color.background.2}}` | `#304050` | ✅ |
| `#c8d8e8` text | `{{color.text.primary}}` | `#c8d8e8` | ✅ |
| `#0090e0` hover border | `{{color.accent}}` | **`#00b4d8`** | ❌ |
| `#00d0ff` XVTR text | `{{color.accent.bright}}` | **`#00c8f0`** | ❌ |

And it stops three-quarters of the way: the fourth identical stylesheet, the separate XVTR panel at `SpectrumOverlayMenu.cpp:3242`, is left hardcoded. So after this the XVTR panel and the band panel render differently from each other in both themes.

Moving this file onto theme tokens is a change I'd expect the project to want — it is the direction the colour ratchet and `tools/check_theme_seed.py` push. That is an argument for giving it its own PR where it can be done completely and reviewed as what it is, not for landing three-quarters of it inside a highlight feature.

## Findings — not blocking

**3. The highlight's band precedence is the inverse of the selection path's.** *(inline: `SpectrumOverlayMenu.cpp:3366`)* Here the declared-range loop runs after the XVTR loop and overwrites `activeBand`, so **declared range wins**. In `MainWindow::selectBand()` — the handler for clicking these very buttons — `XvtrPolicy::resolveBandStackKey()` runs first and the declared-band fallthrough only executes `if (stackKey.isEmpty())`, so **XVTR wins**. That file states the rule explicitly: *"a user XVTR labelled e.g. `2m` out-ranks a gateway's declared `2m` … consistent with the XVTR-priority precedent already codified for native bands (#2342)."* On a radio with a declared band overlapping a user XVTR, the highlighted button and the button that actually gets selected are different ones — which defeats the point of the highlight. Two-line reorder.

**4. The XVTR match is a ±500 kHz window around a single frequency.** *(same inline)* `std::abs(freq - xvtr.rfFreqMhz) < 0.5` is an uncommented magic constant, and the very next loop shows the alternative — `XvtrBand` neighbours a `DeclaredBandRange` that carries real `lowHz`/`highHz` bounds. A transverter covering 144–148 MHz highlights only within 500 kHz of its configured RF frequency, and two XVTRs less than 1 MHz apart both match with first-wins arbitration.

## Nits

- `updateActiveBandHighlight()` runs on every `frequencyChanged`, which fires continuously during a VFO drag, and each call re-derives the band and touches every button. `setChecked()` is cheap when unchanged, so this is not a measurable problem — but caching the last resolved `activeBand` and returning early would make it obviously free.
- Moving `wfSchemeName()` into the header as `inline` is fine and correctly disclosed. Worth a one-line comment saying *why* it lives in the header, so nobody moves it back and breaks the test link.

## Verified vs. read

**Built and ran:** the full app target and the new test. **Measured:** the token values by resolving `default-dark.json` and `default-light.json` through their `primitives` maps, and the contrast figures in blocker 1 from those; confirmed both files carry only a `root` scope; confirmed on `main` that none of the four `rgba(30, 40, 55, 220)` styles used a colour token, so the light-theme regression tracks to this diff. **Read:** the precedence comparison against `MainWindow::selectBand()` in finding 3. **Not verified:** I did not run the app in light mode to look at it — the contrast numbers are computed from the theme files, not observed on screen. If you have a light-mode screenshot that contradicts them, I would rather see it than be right.

File: src/gui/SpectrumOverlayMenu.cpp [hunk 7]
```diff
@@ -264,6 +264,30 @@ static const QString kMenuBtnActive =
     "border: 1px solid #0090e0; border-radius: 2px; "
     "color: #ffffff; font-size: 11px; font-weight: bold; }";
 
+static const QString kBandBtnStyle =
+    "QPushButton { background: rgba(30, 40, 55, 220); "
+    "border: 1px solid #304050; border-radius: 3px; "
+    "color: #c8d8e8; font-size: 11px; font-weight: bold; }"
+    "QPushButton:hover { background: rgba(0, 112, 192, 180); "
+    "border: 1px solid #0090e0; }"
+    "QPushButton:checked { background: #0070c0; color: #ffffff; "
```
Comment (jensenpat) [inline src/gui/SpectrumOverlayMenu.cpp]: **Blocker 1 — approval boundary remains:** these new `:checked` colours change the visual design of existing controls. `GOVERNANCE.md` says *any change to visual design — colors, fonts, spacing, theme, icons* requires an approved RFC, and this PR has neither an RFC nor a linked bug defining the intended repair. Please link the approved decision before merge.

File: src/gui/SpectrumOverlayMenu.cpp [hunk 8]
```diff
@@ -3317,6 +3331,61 @@ void SpectrumOverlayMenu::applyTuningRangeToBandButtons()
...
+            if (entry.button) {
+                QSignalBlocker b(entry.button);
+                entry.button->setChecked(false);
+            }
+        }
+        return;
+    }
+
+    const double freq = m_slice->frequency();
+    QString activeBand = BandSettings::bandForFrequency(freq);
```
Comment (jensenpat) [inline src/gui/SpectrumOverlayMenu.cpp]: **Blocker 2 — still US-only:** resolving the earlier thread did not change this call. `BandSettings::bandForFrequency()` uses `BandDefs.h`, while `AGENTS.md` explicitly requires new band-edge consumers to read the active regional plan through `BandPlanManager`. Please route plan-aware classification into this menu and add a non-US boundary case (for example, a Region 1 frequency outside its allocation but inside the US table).

**Review by jensenpat — CHANGES_REQUESTED** [review]
The revised head fixes the earlier live-theme regression by reverting the partial token migration, and its XVTR precedence now matches the selection path. It still is not merge-ready: there is no linked issue or approved RFC for the new checked-state visual treatment, native-band classification still uses the US-only table, and the exact head conflicts with current `main` in `tests/tests.cmake`.

### Scope

| File/group | Claimed by title/body? | Verdict |
|---|---|---|
| `SpectrumOverlayMenu.{h,cpp}` active-band state and checked styles | Yes | In scope, but blocked by findings 1 and 2 |
| `SpectrumOverlayMenu.cpp` colour-picker/range-label style deduplication | No | Small behavior-neutral cleanup, but undisclosed and not needed for the feature |
| `SpectrumWidget.{h,cpp}` inline relocation | Yes | In scope as the focused-test link seam |
| `tests/spectrum_overlay_band_highlight_test.cpp`, `tests/tests.cmake` | Yes | In scope and socket-free, but regional-plan coverage is absent; registration conflicts with current `main` |

The feature changes are otherwise coherent, but the body should disclose or drop the incidental stylesheet deduplication. No `CHANGELOG.md` change is present.

### Blockers

1. The feature adds a checked-state colour treatment to existing controls without the approved RFC required by `GOVERNANCE.md` for any visual-design change. There is also no linked issue establishing it as a bug fix. See inline comment 1.
2. `updateActiveBandHighlight()` still calls `BandSettings::bandForFrequency()`. `AGENTS.md` says new band-edge consumers must use the active `BandPlanManager`, because `BandDefs.h` is ARRL/US-only. The resolved prior thread did not correspond to a code change, and the focused test still has no non-US edge case. See inline comment 2.
3. Exact-head merge simulation against refreshed `origin/main` fails with a content conflict in `tests/tests.cmake`. Rebase/merge current `main`, preserve both the new highlight test and current-main registrations/default timeout logic, then rerun checks and review on the new SHA.

### Nits (non-blocking)

- Disclose or drop the unrelated colour-picker/range-label stylesheet deduplication.
- The test still does not exercise the PR body's radio-declared-range claim.

### Verification

Reviewed signed head `9cf86f96778e53eeca297f2753c4ccaf3c49c60e` against refreshed `origin/main` `bc3458509b4081b401a17ff510d23e297b9e0ec1`. GitHub's four checks passed on the PR head, all six review threads are resolved, and the added test is a headless `QApplication` test with no socket fixture. The exact colour audit passes (`unique_colours +0`, `total_references -3`, `setStyleSheet +0`). `git merge-tree --write-tree origin/main origin/pr-5236` reports the `tests/tests.cmake` conflict, so I stopped before building or running tests.

**Review by jensenpat — APPROVED** [review]
Approved on exact head 41863ce47a809076c1f364bbf784e199b8897550 after maintainer-directed repair. The branch is rebased onto current main; the selected band uses a high-contrast 2px border and theme-resolved surface/text colors that update live; native-band matching is constrained by the active regional BandPlanManager; XVTR/declared precedence and rebuild safety are covered. Local focused headless test passed 1/1, static audits passed, exact merge simulation is clean, signatures are verified, and all required CI checks are green.
